test/test_automl/test_estimators.py

```diff
@@ -317,6 +320,125 @@ def test_cv_results(tmp_dir):
     assert hasattr(cls, 'classes_')
 
 
+@pytest.mark.parametrize('estimator_type,dataset_name', [
+    (AutoSklearnClassifier, 'iris'),
+    (AutoSklearnRegressor, 'boston')
+])
+def test_leaderboard(
+    tmp_dir: str,
+    estimator_type: Type[AutoSklearnEstimator],
+    dataset_name: str
+):
+    # Comprehensive test tasks a substantial amount of time, manually set if
+    # required.
+    MAX_COMBO_SIZE_FOR_INCLUDE_PARAM = 2  # [0, len(valid_columns) + 1]
+
+    X_train, Y_train, _, _ = putil.get_dataset(dataset_name)
+    model = estimator_type(
+        time_left_for_this_task=30,
+        per_run_time_limit=5,
+        tmp_folder=tmp_dir,
+        seed=1
+    )
+    model.fit(X_train, Y_train)
+
+    valid_columns = [
+        "rank", "ensemble_weight", "type", "cost", "duration", "config_id",
+        "train_loss", "seed", "start_time", "end_time", "budget", "status",
+        "data_preprocessors", "feature_preprocessors", "balancing_strategy",
+        "config_origin"
+    ]
+    simple_columns = ["rank", "ensemble_weight", "type", "cost", "duration"]
+    detailed_columns = valid_columns
+
+    # Create a dict of all possible param values for each param
+    # with some invalid one's of the incorrect type
+    include_combinations = itertools.chain(
+        itertools.combinations(valid_columns, item_count)
+        for item_count in range(1, MAX_COMBO_SIZE_FOR_INCLUDE_PARAM)
+    )
+    valid_params = {
+        'detailed': [True, False],
+        'ensemble_only': [True, False],
+        'top_k': [-10, 0, 1, 10, 'all'],
+        'sort_by': [*valid_columns, 'invalid'],
+        'sort_order': ['ascending', 'descending', 'invalid', None],
+        'include': itertools.chain([None, 'invalid', 'type'], include_combinations),
+    }
+
+    # Create a generator of all possible combinations of valid_params
+    params_generator = iter(
+        dict(zip(valid_params.keys(), param_values))
+        for param_values in itertools.product(*valid_params.values())
+    )
+
+    for params in params_generator:
+
+        # Invalid top_k should raise an error, is a positive int or 'all'
+        if not (params['top_k'] == 'all' or params['top_k'] > 0):
+            with pytest.raises(ValueError):
+                model.leaderboard(**params)
+
+        # Invalid sort_by column
+        # Can't sort_by rank as rank is based on the sort criteria
+        elif (
+            params['sort_by'] == 'rank'
```

Comment:
Actually, rank would not be based on the sort criterium, but on the cost -> maybe we should leave it in?